pandas/core/resample.py

```diff
@@ -1552,11 +1579,18 @@ def _get_period_bins(self, ax):
         end = ax.max().asfreq(self.freq, how="end")
         bin_shift = 0
 
-        # GH 23882
-        if self.base:
-            # get base adjusted bin edge labels
+        if isinstance(self.freq, Tick):
+            # GH 23882 & 31809: get adjusted bin edge labels with `origin`
+            # and `origin` support. This call only makes sense if the freq is a
+            # Tick since offset and origin are only used in those cases.
+            # Not doing this check could create an extra empty bin.
             p_start, end = _get_period_range_edges(
-                start, end, self.freq, closed=self.closed, base=self.base
+                start,
+                end,
+                self.freq,
+                closed=self.closed,
+                origin=self.origin,
+                offset=self.offset,
```

Comment:
FWIW: I changed this:

```python
if self.origin is not None or self.offset is not None:
```

into

```python
if isinstance(self.freq, Tick):
```

When I realised that it makes only sense of doing this when freq was a Tick. However, I thought that this piece of code would have been quite similar, and that the `if` was done only for perfomance reason, but it seems that one `resample` test is failing if we always enter in the `if`: `pandas/tests/resample/test_period_index.py:210`.

```python
pandas/tests/resample/test_period_index.py:210 (TestPeriodIndex.test_resample_count[2M-expected_vals1])
self = <pandas.tests.resample.test_period_index.TestPeriodIndex object at 0x7f724082c520>
freq = '2M', expected_vals = [60, 40]

    @pytest.mark.parametrize(
        "freq,expected_vals", [("M", [31, 29, 31, 9]), ("2M", [31 + 29, 31 + 9])]
    )
    def test_resample_count(self, freq, expected_vals):
        # GH12774
        series = Series(1, index=pd.period_range(start="2000", periods=100))
        result = series.resample(freq).count()
        expected_index = pd.period_range(
            start="2000", freq=freq, periods=len(expected_vals)
        )
        expected = Series(expected_vals, index=expected_index)
>       tm.assert_series_equal(result, expected)
E       AssertionError: Series are different
E       
E       Series length are different
E       [left]:  3, PeriodIndex(['2000-01', '2000-03', '2000-05'], dtype='period[2M]', freq='2M')
E       [right]: 2, PeriodIndex(['2000-01', '2000-03'], dtype='period[2M]', freq='2M')
```

We have an empty bin that should not exists :man_shrugging:.

When I looked the code, it just seems that `_get_period_range_edges` did not handle well frequencies that are not Ticks.